controllers/mysqlcluster_controller.go

```diff
@@ -260,6 +296,101 @@ func generateRandomBytes(n int) ([]byte, error) {
 	return ret, nil
 }
 
+func mergeMycnf(cnf1, cnf2 map[string]string) {
+	for k, v := range cnf2 {
+		cnf1[normalizeConfKey(k)] = v
+	}
+}
+
+func normalizeConfKey(k string) string {
+	return strings.ReplaceAll(k, "-", "_")
+}
+
+func (r *MySQLClusterReconciler) createOrUpdateConfigMap(ctx context.Context, log logr.Logger, cluster *mocov1alpha1.MySQLCluster) error {
+	cm := &corev1.ConfigMap{}
+	cm.SetNamespace(cluster.Namespace)
+	cm.SetName(cluster.Name)
+
+	op, err := ctrl.CreateOrUpdate(ctx, r.Client, cm, func() error {
+		cm.Labels = map[string]string{
+			appNameKey: fmt.Sprintf("%s-%s", appName, cluster.Name),
+		}
+
+		conf := make(map[string]mysqlConf)
+		conf["mysqld"] = make(mysqlConf)
+		mergeMycnf(conf["mysqld"], defaultMycnf)
+
+		if cluster.Spec.MySQLConfigMapName != nil {
+			cm := &corev1.ConfigMap{}
+			err := r.Get(ctx, client.ObjectKey{Namespace: cluster.Namespace, Name: *cluster.Spec.MySQLConfigMapName}, cm)
+			if err != nil {
```

Comment:
Don't you need to check it with [`apierrors.IsNotFound`](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors?tab=doc#IsNotFound)?

Reply:
If a cm doesn't exist in a cluster namespace,  the controller only returns an error. Is there anything it would care about?

Reply:
Ah, okay.  only when `cluster.Spec.MySQLConfigMapName` is not nil.